ibis/backends/dask/udf.py

```diff
@@ -0,0 +1,121 @@
+import itertools
+
+import dask.dataframe as dd
+import dask.dataframe.groupby as ddgb
+import pandas
+
+import ibis.client
+import ibis.expr.operations as ops
+import ibis.udf.vectorized
+from ibis.backends.pandas.client import ibis_dtype_to_pandas
+from ibis.backends.pandas.udf import nullable, udf  # noqa
+
+from .dispatch import execute_node, pre_execute
+from .execution.util import assert_identical_grouping_keys, make_selected_obj
+
+
+@pre_execute.register(ops.ElementWiseVectorizedUDF)
+@pre_execute.register(ops.ElementWiseVectorizedUDF, ibis.client.Client)
+def pre_execute_elementwise_udf(op, *clients, scope=None, **kwargs):
+    """Register execution rules for elementwise UDFs.
+    """
+    input_type = op.input_type
+
+    # definitions
+
+    # Define an execution rule for elementwise operations on a
+    # grouped Series
+    nargs = len(input_type)
+
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF,
+        *(itertools.repeat(ddgb.SeriesGroupBy, nargs)),
+    )
+    def execute_udf_node_groupby(op, *args, **kwargs):
+        func = op.func
+
+        # all grouping keys must be identical
+        assert_identical_grouping_keys(*args)
+
+        # we're performing a scalar operation on grouped column, so
+        # perform the operation directly on the underlying Series
+        # and regroup after it's finished
+        args_objs = [make_selected_obj(arg) for arg in args]
+        groupings = args[0].index
+        return dd.map_partitions(func, *args_objs).groupby(groupings)
```

Comment:
Why do we need a groupby for element wise UDF?

Reply:
https://github.com/ibis-project/ibis/pull/2648#discussion_r592864535

Reply:
Sorry I am not familiar with the part of the dask API used here:

- What are the types of `args_objs` here? 
- How does the freeform function dd.map_partitions() work
- Are we returning a ddgb.DataFrameGroupby object from this function? If so, why are we not returning a dd.DataFrame?

Reply:
- `make_selected_obj` in this case returns a `dd.Series`, so `args_objs` is a `Lis[dd.Series]`. 
- My understanding is `map_partitions`  sets up all the machinery to apply the func on the args passed. It knows about which args are partitionable vs which are scalars and pushes that down accordingly. Is that what you're asking?
- This matches what pandas does in https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/udf.py#L188

Reply:
Oh I see. Actually the pandas logic looks a bit puzzling as well.

For my knowledge, when this returns a SeriesGroupBy object, how does the down stream node consume it? i.e., is there a step somewhere else to turn this back to a dd.DataFrame if the down stream operation is sth like mutate?

Reply:
It looks like generally an aggregation is performed on it afterwards thus making it a ungrouped series post the aggregation. 
For example: `t.groupby(t.key).aggregate(my_add=my_add(t.b, t.c).sum())`